### PR TITLE
Contributing Markdown Refresh

### PR DESCRIPTION
**Pull request by itz-Me-Pj**
Updated Contributing.md file to be more beginner friendly for contributions.
[#4132 ](https://github.com/reduxjs/redux-toolkit/issues/4132)

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -8,25 +8,57 @@ Before opening an issue, please search the [issue tracker](https://github.com/re
 
 Please ask any general and implementation specific questions on [Stack Overflow with a Redux Toolkit tag](http://stackoverflow.com/questions/tagged/redux-toolkit?sort=votes&pageSize=50) for support.
 
-## New Features
+We ask you to do this because StackOverflow has a much better job at keeping popular questions visible. Unfortunately good answers get lost and outdated on GitHub.
 
-Please open an issue with a proposal for a new feature or refactoring before starting on the work. We don't want you to waste your efforts on a pull request that we won't want to accept.
+Some questions take a long time to get an answer. If your question gets closed or you don't get a reply on StackOverflow for longer than a few days, we encourage you to post an issue linking to your question. We will close your issue but this will give people watching the repo an opportunity to see your question and reply to it on StackOverflow if they know the answer.
+
+Please be considerate when doing this as this is not the primary purpose of the issue tracker.
```
Comment (EskiMojo14) [inline CONTRIBUTING.md:15]: i don't agree with this - they should either open a discussion or try the reactiflux  #redux channel, not use the issue tracker

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -8,25 +8,57 @@ Before opening an issue, please search the [issue tracker](https://github.com/re
...
+
+### Help Us Help You
+
+On both websites, it is a good idea to structure your code and question in a way that is easy to read to entice people to answer it. For example, we encourage you to use syntax highlighting, indentation, and split text in paragraphs.
```
Comment (EskiMojo14) [inline CONTRIBUTING.md:19]:
```suggestion
On both websites, it is a good idea to structure your code and question in a way that is easy to read to help people to answer it. For example, we encourage you to use syntax highlighting, indentation, and split text in paragraphs.
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -8,25 +8,57 @@ Before opening an issue, please search the [issue tracker](https://github.com/re
...
+
+Please keep in mind that people spend their free time trying to help you. You can make it easier for them if you provide versions of the relevant libraries and a runnable small project reproducing your issue. You can put your code on [JSBin](https://jsbin.com) or, for bigger projects, on GitHub. Make sure all the necessary dependencies are declared in `package.json` so anyone can run `npm install && npm start` and reproduce your issue.
 
 ## Getting started
 
 Visit the [Issue tracker](https://github.com/reduxjs/redux-toolkit/issues) to find a list of open issues that need attention.
 
+### Bugs and Improvements
+
+We use the [Issue tracker](https://github.com/reduxjs/redux-toolkit/issues) to keep track of bugs and improvements to redux-toolkit itself, its examples, and the documentation. We encourage you to open issues to discuss improvements, architecture, theory, internal implementation, etc. If a topic has been discussed before, we will ask you to join the previous discussion.
```
Comment (EskiMojo14) [inline CONTRIBUTING.md:29]:
```suggestion
We use the [Issue tracker](https://github.com/reduxjs/redux-toolkit/issues) to keep track of bugs and improvements to Redux Toolkit itself, its examples, and the documentation. We encourage you to open issues to discuss improvements, architecture, theory, internal implementation, etc. If a topic has been discussed before, we will ask you to join the previous discussion.
```

File: packages/toolkit/README.md [hunk 4]
```diff
@@ -48,6 +48,10 @@ The Redux Toolkit docs are available at **https://redux-toolkit.js.org**, includ
 
 The Redux core docs at https://redux.js.org includes the full Redux tutorials, as well usage guides on general Redux patterns.
 
+## Contributing
+
+Pls refer to our [contributing guide](/CONTRIBUTING.md) to learn about our development process, how to propose bugfixes and improvements, and how to build and test your changes to Redux-toolkit.
```
Comment (EskiMojo14) [inline packages/toolkit/README.md:53]:
```suggestion
Please refer to our [contributing guide](/CONTRIBUTING.md) to learn about our development process, how to propose bugfixes and improvements, and how to build and test your changes to Redux Toolkit.
```

**Review by EskiMojo14 — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (itz-Me-Pj) [reply CONTRIBUTING.md:15]: @EskiMojo14 
Some questions take a long time to get an answer. If your question gets closed or you don't get a reply on StackOverflow for longer than a few days, we encourage you to either open a discussion or try the reactiflux (#redux channel) on discord our redux community will help you resolving the issue.
How does this sound?

Comment (EskiMojo14) [reply CONTRIBUTING.md:15]: yeah, i think:
```suggestion
If your question gets closed or you don't get a reply after a few days, consider opening a [discussion](https://github.com/reduxjs/redux-toolkit/discussions) or joining the [Reactiflux](https://discord.gg/reactiflux) discord server and asking in the #redux channel.
```

**Comment by itz-Me-Pj** [comment]
@EskiMojo14 all the requested changes have been made successfully. 
Do you want me to update the code of conduct doc as well, as it hasn't been updated since a long time and currently the redux code of conduct is following code covenant version 1.4  where as the toolkit is at 1.0?
Also I can add ways you can contribute in the contribution doc by referring them to the right doc folder or the right example folder or the right development block for making successful contributions.

File: packages/toolkit/README.md [hunk 5]
```diff
@@ -48,6 +48,10 @@ The Redux Toolkit docs are available at **https://redux-toolkit.js.org**, includ
 
 The Redux core docs at https://redux.js.org includes the full Redux tutorials, as well usage guides on general Redux patterns.
 
+## Contributing
```
Comment (EskiMojo14) [inline packages/toolkit/README.md:51]: re-reading this, i think it should be moved to the end of the readme, after the sections about what it's for and what it includes

**Comment by EskiMojo14** [comment]
code of conduct i don't really know about - for now i think it should be left as is.

i don't really understand what you mean regarding referring to the "correct doc folder" etc

**Comment by itz-Me-Pj** [comment]
@EskiMojo14 by referring to doc or example folder I meant to add something like:-
**Ways you can contribute**
There are several ways you can contribute to the repository instead of developing a feature or fixing a bug you can contribute by updating the [documentation](https://github.com/reduxjs/redux-toolkit/tree/master/docs) for a specific feature or implementation. Or, you can add [examples](https://github.com/reduxjs/redux-toolkit/tree/master/examples) of any 3rd party implementation like graphQL or React (just an example).

**Comment by EskiMojo14** [comment]
sure, can't hurt - those sentences are quite long so some punctuation would be good but otherwise what you have there is fine

**Comment by itz-Me-Pj** [comment]
@EskiMojo14 Added the final changes to the contribution doc. Pls review and tell if anything else is left to add.
Also I was thinking to work on rewriting the documentation in [issues](https://github.com/reduxjs/redux/issues?page=1&q=is%3Aissue+is%3Aopen) that are open since a long time in the core redux repo. Pls have a look and suggest if they can be worked upon or not.

**Comment by EskiMojo14** [comment]
thanks!

with regards to core docs, i would need to know specifically which one you want to write, but generally if it's still open we'd welcome a PR for it

**Pull request merged**